doc/source/overview/config_files.md

```diff
@@ -0,0 +1,753 @@
+`preprocess_fn`, which themselves rely on other artefacts. TOML files are highly flexible when it comes to specifying 
+dictionaries; each section in a TOML file, demarcated by section headers enclosed in [ ] brackets, is actually a Python 
+dictionary (when the file is parsed). For example, the `preprocess_fn` can be specified as a dictionary with:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "function.dill"
+```
+
+Alternatively, the following are also equivalent:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn = {src="function.dill"}
+```
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn.src = "function.dill"
+```
+
+The above schemas are not strictly necessary in this case, since `preprocess_fn` could simply be specified with
+`preprocess_fn = "function.dill"` here. However, the need to specify fields as dictionaries will become clear in the 
+following sections, where the layouts of the various possible artefact dictionaries are documented.
+
+#### Preprocessing function
+
+Simple `preprocess_fn`'s can be specified directly as a serialized dill file or via a function registry. If additional 
+arguments are to be passed at call time, the `preprocess_fn` can instead be specified as a dictionary:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "function.dill"
+kwargs = {kwarg1="a", kwarg2=1.5, kwarg3=true}
+```
+
+In this case, the `src` field can be a locally stored `.dill` file or a function registry, and `kwargs` is a 
+dictionary of keyword arguments to be passed to the function. As noted previously, there is considerable flexibility 
+with regard to how these fields are specified. For example, the following is equivalent:
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+preprocess_fn.src = "function.dill"
+
+[preprocess_fn.kwargs]
+kwarg1 = "a"
+kwarg2 = 1.5
+kwarg3 = true
+```
+
+A special (but common!) case for `preprocess_fn` is to use the Alibi Detect utility function 
+[preprocess_drift](../api/alibi_detect.cd.tensorflow.preprocess.rst) (or its pytorch equivalent). This is a utility 
+function to incorporate a user-supplied model, batch preprocessing function, and/or tokenizer into the preprocessing 
+step of a drift detector. It is pre-registered, so can be specified with 
+`src="@cd.tensorflow.preprocess.preprocess_drift"`, and the rest of the fields in `preprocess_fn` are then the 
+args/kwargs to pass to `preprocess_drift`. For example: 
+
+<p class="codeblock-label">config.toml (excerpt)</p>
+
+```toml
+[preprocess_fn]
+src = "@cd.tensorflow.preprocess.preprocess_drift"
+batch_size = 32
+preprocess_batch_fn = "batch_fn.dill"
+```
+
+#### Models 
+
+A `model` dictionary may be used to specify a model within a `preprocess_fn`, or a model for detectors such as the
+[ClassifierDrift](../api/alibi_detect.cd.classifier.rst) detector. The possible fields are:
```

Comment:
Hmm.. does this mean that a top-level `model` field can be parsed either as part of `preprocess_fn.model` or as a `model` going straight into the detector? What if both are needed? I personally don't like this ambiguity...

Reply:
Yeh the same `ModelConfig` is used to specify a top-level `model` field to be passed in as an argument to a detector such as `ClassifierDrift`, but it can also be used to define a model to be passed into the `preprocess_fn` (but the latter would require `preprocess_fn.model` to be set, not `model`). For example, something like the below should be valid:

```toml
name = "ClassifierDrift"
x_ref = "x_ref.npy"

[model]
type = "custom"
src = "model"

[preprocess_fn]
src = "@cd.tensorflow.preprocess.preprocess_drift"

[preprocess_fn.model]
type = "HiddenOutput"
src = "preprocess_fn/model"
```
The same applies for the `DeepKernel.proj` field, which is expected to be a TF/torch `model`.

The slight ambiguity here comes from the fact that I've tried to keep field names in sync with detector and function arg/kwarg names wherever possible, so that we don't have to maintain a mapping between config fields and detector/function args/kwarg (and a user can go to the relevant api docs for more info on a field). e.g. all the top-level fields in the config (except for `version` and `config_spec`) should match the names of a spec'd detectors args/kwargs. Also a similar story when `preprocess_drift` used for the `preprocess_fn`...